### PR TITLE
[WIP] libp2p extra logging on internode communication #2699

### PR DESCRIPTION
**Pull request by Karrenbelt**
Additional logging ported from https://github.com/fetchai/agents-aea/pull/2699

Should not be merged before open-aea is updated, because tests that checks for code similarity will otherwise fail

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-go@master
         with:
-          go-version: '^1.14.0'
+          go-version: "1.17.7"
```
Comment (Karrenbelt) [inline .github/workflows/workflow.yml:22]: set to 1.17.7 throughout, which is what we use in open-aea workflow as well

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -19,14 +19,14 @@ jobs:
...
       - name: Install dependencies (ubuntu-latest)
         run: |
           sudo apt-get update --fix-missing
           sudo apt-get autoremove
           sudo apt-get autoclean
       - name: Golang code style check (libp2p_node)
-        uses: golangci/golangci-lint-action@v1
+        uses: golangci/golangci-lint-action@v3
```
Comment (Karrenbelt) [inline .github/workflows/workflow.yml:29]: we use @v3 in workflow in [open-aea workflow](https://github.com/valory-xyz/open-aea/blob/d72f99a960ce1f81c2083206e00acf11b83e4aa4/.github/workflows/workflow.yml#L80)

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -45,13 +45,13 @@ jobs:
...
-    - uses: actions/setup-go@master
+    - uses: actions/setup-go@v3
       with:
-        go-version: '^1.14.0'
+        go-version: "1.17.7"
     - if: matrix.os == 'macos-latest'
       working-directory: .
       run: |
-         export LINKPATH=$GOROOT/pkg/tool/darwin_amd64/link
+         export LINKPATH=`go env GOTOOLDIR`/link
```
Comment (8ball030) [inline .github/workflows/workflow.yml:54]: Whats this?

Comment (8ball030) [reply .github/workflows/workflow.yml:54]: Can you please give me a refference on this as i am unsure what it is for

**Review by 8ball030 — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**